chain/chunks/src/test/multi.rs

```diff
@@ -187,10 +191,14 @@ fn test_distribute_chunk_track_all_shards() {
     });
     // Two network rounds is enough because each node should have
     // forwarded the parts to those block producers that need them.
-    // TODO: after phase 2, we will need a longer delay because validators
-    // that don't track the shard will not get forwards. We may also need
-    // to add the periodic resending handler.
-    test.run_for(NETWORK_DELAY * 2);
+    // However, after SingleShardTracking protocol upgrade, we need a longer
+    // delay because validators that don't track the shard will not get
+    // parts forwarded to them.
```

Comment:
Could you add what is the longest flow of messages so that 4 hops are required? I meditated on https://pagodaplatform.atlassian.net/wiki/spaces/EAP/pages/136511489/How+Chunks+Are+Distributed but it didn't immediately follow to me why multiple messages are needed.

Reply:
Expanded the comments. Also, you can look at the log visualization of the test, instructions here https://github.com/near/nearcore/blob/master/tools/debug-ui/README.md#how-to-use-the-testloop-log-visualizer

this makes it a lot clearer what's going on.